internal/scheduler/scheduler.go

```diff
@@ -228,23 +228,25 @@ func (s *Scheduler) runJob(ctx context.Context, r *job.Run) error {
 	jobContext, rj.cancelCtx = context.WithCancel(ctx)
 
 	go func() {
-		defer rj.cancelCtx()
 		runErr := j.Run(jobContext)
-		var updateErr error
+		// Job has ended, cancel job run specific context to clean up
+		rj.cancelCtx()
```

Comment:
I think the original `defer rj.cancelCtx()` was cleaner and more idiomatic. Is the `defer` causing a problem?

Reply:
No it is not the issue.  I changed it back was on the fence about this change after I did it.  I did it because I wanted to be clear that the `jobContext` should not be used outside of the `Run`.